pkg/loadbalancer/loadbalancer.go

```diff
@@ -186,6 +187,40 @@ func (si K8sServiceInfo) IsExternal() bool {
 	return len(si.Selector) == 0
 }
 
+// Equals returns true if K8sServiceInfo is considered equal to the given
+// k8sServiceInfo.
+// Parameters:
+//  * o K8sServiceInfo to be compared with.
+func (si *K8sServiceInfo) Equals(o *K8sServiceInfo) bool {
+	if si == nil {
+		if o == nil {
```

Comment:
It seems inconsistent to not handle `o == nil && si != nil`. This applies to all functions.